adi/cn0554.py

```diff
@@ -0,0 +1,148 @@
+# are permitted provided that the following conditions are met:
+#     - Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     - Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import adi
+
+
+class cn0554:
+    """ CN0554 Mixed Signal Raspberry Pi Hat """
+
+    _in_range_available = ["+/-13.75", "+27.5", "+2.5"]
+    _out_reference_available = [4.096, 2.5]
+    _out_channels = 16
+    _in_channels = 8
+
+    def __init__(self, uri="ip:analog.local"):
+        self.scale_res = 11  # Based on resistors used in input
+        self.adc = adi.ad7124(uri=uri)
+        self.adc._rx_unbuffered_data = False
+        self.adc._rx_output_type = "SI"
+        self.data_points = 1024
+        self.adc.sample_rate = 19200
+        self.adc_in_channels = []
+
+        for in_ch in self.adc.channel:
+            in_name = "voltage"
+            for char in in_ch.name:
+                if char.isnumeric():
+                    in_name += char
+
+            setattr(
+                self, in_name + "_in_range", self._in_range_available[0]
+            )  # default range of HW
+
+            self.adc_in_channels.append(in_name)
+
+        self.dac = adi.ltc2688(uri=uri)
+        self.dac_out_channels = self.dac.channel_names
+        self.out_reference = self._out_reference_available[0]
```

Comment:
Put in class definition above with None as default value with docstring and annotations